app/controllers/components/course/discussion/topics_component.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+class Course::Discussion::TopicsComponent < SimpleDelegator
+  include Course::ControllerComponentHost::Component
+
+  def self.display_name
+    I18n.t('components.discussion.topics.name')
+  end
+
+  def sidebar_items
+    main_sidebar_items + settings_sidebar_items
+  end
+
+  def settings
+    @settings ||= Course::Discussion::TopicSettings.new(current_course.settings(:discussion_topics))
+  end
+
+  private
+
+  def main_sidebar_items
+    [
+      {
+        key: :discussion_topics,
+        title: settings.title || t('course.discussion.topics.sidebar_title'),
+        weight: 5,
+        path: sidebar_path,
+        unread: unread_count
+      }
+    ]
+  end
+
+  def settings_sidebar_items
+    [
+      {
+        title: t('course.discussion.topics.sidebar_title'),
+        type: :settings,
+        weight: 4,
+        path: course_admin_topics_path(current_course)
+      }
+    ]
+  end
+
+  def sidebar_path
+    if staff_with_students?
+      my_students_pending_course_topics_path(current_course)
+    elsif staff?
+      pending_course_topics_path(current_course)
+    else
+      course_topics_path(current_course)
+    end
+  end
+
+  def unread_count
+    if staff_with_students?
+      my_students_unread_count
+    elsif staff?
+      all_unread_count
+    else
+      0
+    end
+  end
+
+  def staff_with_students?
+    current_course_user && current_course_user.staff? && !current_course_user.my_students.empty?
+  end
+
+  def staff?
+    # TA is the course staff, but cannot manage course.
+    (current_course_user && current_course_user.staff?) || can?(:manage, current_course)
```

Comment:
Actually global admins need not have the bubble at the sidebar - I think it is unrelated to them since they are not enrolled in the course. May I suggest that we remove that segment with `can?`? Sorry for flip flopping!